src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -644,10 +643,9 @@ ImmutableArray<byte> computeContentHash()
                         hash.Append(MemoryMarshal.AsBytes(charBuffer.AsSpan(0, charsToCopy)));
                     }
 
-                    // Switch this to ImmutableCollectionsMarshal.AsImmutableArray(hash.GetHashAndReset()) when we move to S.C.I v8.
-                    Span<byte> destination = stackalloc byte[128 / 8];
-                    hash.GetHashAndReset(destination);
-                    return destination.ToImmutableArray();
+                    // GetHashAndReset returns a freshly allocated array so it is safe to convert
+                    // to ImmutableArray inline.
+                    return ImmutableCollectionsMarshal.AsImmutableArray(hash.GetHashAndReset());
```

Comment:
Consider adding similar comment to other places where ImmutableCollectionsMarshal is used.